### ISSUE TITLE
Don't trigger missing-function-docstring with @typing.overload

### ISSUE THREAD

**Issue by dseomn** [PyCQA/pylint#3350]
### Is your feature request related to a problem? Please describe
Linting this module gives `missing-function-docstring` for the two `@typing.overload` functions:

```python
import typing


@typing.overload
def f(x: str) -> bytes:
    pass


@typing.overload
def f(x: bytes) -> str:
    pass


def f(x):
    """This is a docstring."""
    if isinstance(x, str):
        return x.encode()
    else:
        return x.decode()
```

### Describe the solution you'd like
I think docstrings should only go on the function implementation, not the `@typing.overload`-decorated stubs, so it would be nice if pylint didn't trigger `missing-function-docstring` for those.

### Additional context
https://www.python.org/dev/peps/pep-0484/#function-method-overloading
